app/src/main/java/com/zeapo/pwdstore/crypto/PgpActivity.kt

```diff
@@ -100,6 +102,12 @@ class PgpActivity : AppCompatActivity(), OpenPgpServiceConnection.OnBound {
     }
     private var mServiceConnection: OpenPgpServiceConnection? = null
 
+    private val receiver = object : BroadcastReceiver() {
+        override fun onReceive(p0: Context?, p1: Intent?) {
+            delayTask?.doOnPostExecute()
```

Comment:
We should try to get rid of `DelayTask` entirely

Reply:
Yeah but we would still need some abstraction to manage the service and related tasks, also delay task is deeply coupled with the existing code, removing it might cause a lot issues.

Reply:
Well it is not as coupled as I thought, that abstraction still seems important, your call tho